src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheck.java

```diff
@@ -0,0 +1,271 @@
+import net.sf.saxon.sxpath.XPathEvaluator;
+import net.sf.saxon.sxpath.XPathExpression;
+import net.sf.saxon.trans.XPathException;
+
+/**
+ * <p>
+ * Evaluates Xpath query and report violation on all matching AST nodes. Current check allows
+ * user to implement custom checks using Xpath. If Xpath query is not specified explicitly,
+ * then check does nothing.
+ * </p>
+ * <p>
+ * Please read more about Xpath syntax at <a href="https://www.w3schools.com/xml/xpath_syntax.asp">
+ * W3Schools Xpath Syntax</a>. Information regarding Xpath functions can be found at
+ * <a href="https://developer.mozilla.org/en-US/docs/Web/XPath/Functions">XSLT/XPath Reference</a>.
+ * Note, that <b>@text</b> attribute can used only with token types that are listed in
+ * <a href="https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/utils/XpathUtil.java#L101">
+ *     XpathUtil</a>.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code query} - Xpath query.
+ * Type is {@code java.lang.String}.
+ * Default value is {@code ""}.
+ * </li>
+ * </ul>
+ * <p>
+ * Checkstyle provides <a href="https://checkstyle.org/cmdline.html">command line tool</a>
+ * and <a href="https://checkstyle.org/writingchecks.html#The_Checkstyle_SDK_Gui">GUI
+ * application</a> with options to show AST and to ease usage of Xpath queries.
+ * </p>
+ * <p><b>-T</b> option prints AST tree of the checked file.</p>
+ * <pre>
+ * $ java -jar checkstyle-X.XX-all.jar -T Main.java
+ * CLASS_DEF -&gt; CLASS_DEF [1:0]
+ * |--MODIFIERS -&gt; MODIFIERS [1:0]
+ * |   `--LITERAL_PUBLIC -&gt; public [1:0]
+ * |--LITERAL_CLASS -&gt; class [1:7]
+ * |--IDENT -&gt; Main [1:13]
+ * `--OBJBLOCK -&gt; OBJBLOCK [1:18]
+ * |--LCURLY -&gt; { [1:18]
+ * |--METHOD_DEF -&gt; METHOD_DEF [2:4]
+ * |   |--MODIFIERS -&gt; MODIFIERS [2:4]
+ * |   |   `--LITERAL_PUBLIC -&gt; public [2:4]
+ * |   |--TYPE -&gt; TYPE [2:11]
+ * |   |   `--IDENT -&gt; String [2:11]
+ * |   |--IDENT -&gt; sayHello [2:18]
+ * |   |--LPAREN -&gt; ( [2:26]
+ * |   |--PARAMETERS -&gt; PARAMETERS [2:27]
+ * |   |   `--PARAMETER_DEF -&gt; PARAMETER_DEF [2:27]
+ * |   |       |--MODIFIERS -&gt; MODIFIERS [2:27]
+ * |   |       |--TYPE -&gt; TYPE [2:27]
+ * |   |       |   `--IDENT -&gt; String [2:27]
+ * |   |       `--IDENT -&gt; name [2:34]
+ * |   |--RPAREN -&gt; ) [2:38]
+ * |   `--SLIST -&gt; { [2:40]
+ * |       |--LITERAL_RETURN -&gt; return [3:8]
+ * |       |   |--EXPR -&gt; EXPR [3:25]
+ * |       |   |   `--PLUS -> + [3:25]
+ * |       |   |       |--STRING_LITERAL -&gt; "Hello, " [3:15]
+ * |       |   |       `--IDENT -&gt; name [3:27]
+ * |       |   `--SEMI -&gt; ; [3:31]
+ * |       `--RCURLY -&gt; } [4:4]
+ * `--RCURLY -> } [5:0]
+ * </pre>
+ * <p><b>-b</b> option shows AST nodes that match given Xpath query. This command can be used to
+ * validate accuracy of Xpath query against given file.</p>
+ * <pre>
+ * $ java -jar checkstyle-X.XX-all.jar -b Main.java //METHOD_DEF[./IDENT[@text='"sayHello"']]
```

Comment:
`java -jar checkstyle-X.XX-all.jar Main.java -b "//METHOD_DEF[./IDENT[@text='"sayHello"']]"`

Reply:
sorry, my bad

done

Reply:
Please do it like:
`java -jar checkstyle-X.XX-all.jar Main.java -b "//METHOD_DEF[./IDENT[@text='sayHello']]"`

Reply:
-_- sorry, done

Reply:
Still not done, before reply please make sure you see it in github your self

Reply:
Misunderstanding, I was thinking that you are referring the order of options, but only today noticed quotes

Reply:
I hope now it is fixed